discovery_engine/test/integration/migration_test.dart

```diff
@@ -0,0 +1,362 @@
+        url: Uri.parse('http://$source2/foo/index.html'),
+      ),
+    ),
+    Document(
+      documentId: DocumentId(),
+      stackId: StackId.fromString('311dc7eb-5fc7-4aa4-8232-e119f7e80e76'),
+      batchIndex: 3,
+      userReaction: UserReaction.negative,
+      isActive: false,
+      isSearched: false,
+      resource: NewsResource(
+        country: 'US',
+        language: 'en',
+        datePublished: DateTime.utc(2020, 3, 7, 8, 9),
+        image: Uri.parse('http://$source3/foo/imag.png'),
+        rank: 3,
+        score: 3,
+        snippet: 'snippet3',
+        sourceDomain: source3,
+        title: 'title3',
+        topic: 'topic3',
+        url: Uri.parse('http://$source3/foo/index.html'),
+      ),
+    ),
+    Document(
+      documentId: DocumentId(),
+      stackId: StackId.nil(),
+      batchIndex: 4,
+      userReaction: UserReaction.positive,
+      isActive: false,
+      isSearched: true,
+      resource: NewsResource(
+        country: 'US',
+        language: 'en',
+        datePublished: DateTime.utc(2020, 4, 7, 8, 9),
+        image: Uri.parse('http://$source4/foo/imag.png'),
+        rank: 4,
+        score: 4,
+        snippet: 'snippet4',
+        sourceDomain: source4,
+        title: 'title4',
+        topic: 'topic4',
+        url: Uri.parse('http://$source4/foo/index.html'),
+      ),
+    )
+  ];
+
+  group('Hive to Rust migration', () {
+    late LocalNewsApiServer server;
+    late TestEngineData data;
+    DiscoveryEngine? engine;
+
+    setUp(() async {
+      data = await setupTestEngineData(useEphemeralDb: false);
+      server = await LocalNewsApiServer.start();
+
+      EventHandler.registerHiveAdapters();
+      await EventHandler.initDatabase(data.applicationDirectoryPath);
+    });
+
+    tearDown(() async {
+      await engine?.dispose();
+      await server.close();
+      await Directory(data.applicationDirectoryPath).delete(recursive: true);
+    });
+
+    test('works with mostly empty db (a)', () async {
+      await HiveInjection.accessRepositories(data.applicationDirectoryPath,
+          (repos) async {
+        await repos.sourceReactedRepository.save(
+          SourceReacted(source1, true),
+        );
+      });
+
+      engine = await initEngine(data, server.port);
+
+      await HiveInjection.accessRepositories(data.applicationDirectoryPath,
+          (repos) async {
+        expect(repos.engineStateRepository.isEmpty, isTrue);
+        expect(repos.documentRepository.isEmpty, isTrue);
+        expect(repos.activeSearchRepository.isEmpty, isTrue);
+        expect(repos.activeDocumentDataRepository.isEmpty, isTrue);
+        expect(repos.sourceReactedRepository.isEmpty, isTrue);
+        expect(repos.sourcePreferenceRepository.isEmpty, isTrue);
+      });
+    });
+
+    test('works with mostly empty db (b)', () async {
+      await HiveInjection.accessRepositories(data.applicationDirectoryPath,
+          (repos) async {
+        await repos.documentRepository.update(docs[0]);
```

Comment:
🎨 a bit nicer to read, but just a suggestion
```suggestion
        await repos.documentRepository.update(docs.first);
```